framework/src/auxkernels/FunctorElementalAux.C

```diff
@@ -28,19 +28,24 @@ FunctorElementalAuxTempl<is_ad>::validParams()
   InputParameters params = AuxKernel::validParams();
   params.addClassDescription(
       "Evaluates a functor (variable, function or functor material property) on the current "
-      "element. For finite volume, this evaluates the material property at the centroid.");
+      "element or quadrature point.");
   params.addRequiredParam<MooseFunctorName>("functor", "The functor to evaluate");
   params.addParam<MooseFunctorName>("factor", 1, "A factor to apply on the functor");
-
   return params;
 }
 
 template <bool is_ad>
 FunctorElementalAuxTempl<is_ad>::FunctorElementalAuxTempl(const InputParameters & parameters)
   : AuxKernel(parameters),
     _functor(getFunctor<GenericReal<is_ad>>("functor")),
-    _factor(getFunctor<GenericReal<is_ad>>("factor"))
+    _factor(getFunctor<GenericReal<is_ad>>("factor")),
+    _use_qp_arg(dynamic_cast<MooseVariableFE<Real> *>(&_var))
```

Comment:
for CONST MONOMIAL FE we are good right?

Reply:
since the qp is the centroid

Reply:
even if the qp is not the centroid it's fine. E.g. we always evaluate with a quadrature rule based on the highest polynomial basis present in the simulation. So if there are first order Lagrange variables, then we would use a standard 2 point quadrature rule in 1D (and its tensor products in higher-D). Then the final monomial value is handled with the code
```c++
    if (_n_local_dofs == 1) /* p0 */
    {
      ComputeValueType value = 0;
      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
        value += _JxW[_qp] * _coord[_qp] * computeValue();
      value /= (_bnd ? _current_side_volume : _current_elem_volume);
```